circuit-types/src/keychain.rs

```diff
@@ -185,14 +185,14 @@ impl<'de, const KEY_WORDS: usize> Deserialize<'de> for NonNativeKey<KEY_WORDS> {
     }
 }
 
-impl<const KEY_WORDS: usize> From<&NonNativeKey<KEY_WORDS>> for DalekKey {
-    fn from(val: &NonNativeKey<KEY_WORDS>) -> Self {
+impl<const SCALAR_WORDS: usize> From<&NonNativeScalar<SCALAR_WORDS>> for DalekKey {
+    fn from(val: &NonNativeScalar<SCALAR_WORDS>) -> Self {
         let key_bytes = BigUint::from(val).to_bytes_le();
         DalekKey::from_bytes(&key_bytes).unwrap()
     }
 }
 
-impl<const KEY_WORDS: usize> From<DalekKey> for NonNativeKey<KEY_WORDS> {
+impl<const SCALAR_WORDS: usize> From<DalekKey> for NonNativeScalar<SCALAR_WORDS> {
```

Comment:
This can be addressed later, but I believe we will want to remove this `DalekKey` use in favor of an secp256 key, right?

Reply:
ack, yes, but also think it should be addressed later - this was just me find-and-replacing